utils/replicaset/replicaset.go

```diff
@@ -57,6 +64,69 @@ func FindNewReplicaSet(rollout *v1alpha1.Rollout, rsList []*appsv1.ReplicaSet) *
 	return nil
 }
 
+func CheckAndCreateDefaultAntiAffinityRule(RSTemplate *corev1.PodTemplateSpec, rollout v1alpha1.Rollout) {
+	enableAntiAffinity := (rollout.Spec.Strategy.BlueGreen != nil && rollout.Spec.Strategy.BlueGreen.AntiAffinity) || (rollout.Spec.Strategy.Canary != nil && rollout.Spec.Strategy.Canary.AntiAffinity)
```

Comment:
Can we make this into a helper function that breaks it into two if checks? I think that makes it a little more readable since developers don't have to worry about the order of operations